lib/build.gradle.kts

```
@@ -55,6 +67,99 @@ tasks.dokkaJavadoc {
     }
 }
 
+
+tasks.register("generateGuides") {
+    doLast {
+        val kotlinSamplesDir = "${project.projectDir}/src/test/kotlin"
+        val javaSamplesDir = "${project.rootDir}/javademo/src/main/java/com/example/javademo"
+        val guideFile = file("guide.md")
+        val kotlinGuideFile = project.projectDir.resolve("kotlin-guide.md")
+        val javaGuideFile = project.projectDir.resolve("java-guide.md")
+
+        val guideContent = guideFile.readText()
+
+        val kotlinCodeSectionPattern =
+            Regex("""<!-- codeSection\(path="([\w/]+\.kt)", function="([^"]+)", language="kotlin"\) -->""")
+        val javaCodeSectionPattern =
+            Regex("""<!-- codeSection\(path="([\w/]+\.java)", function="([^"]+)", language="java"\) -->""")
+
+        val kotlinCodeBlockPattern = Regex("""```kotlin[\s\S]*?```""")
+        val javaCodeBlockPattern = Regex("""```java[\s\S]*?```""")
+
+        val docsStartPattern = Regex("""^\s*//\s*docsStart\s*$""")
+        val docsEndPattern = Regex("""^\s*//\s*docsEnd\s*$""")
+
+        fun extractCodeSection(path: String, functionName: String, language: String): String {
+            val file = file(path)
+            if (!file.exists()) {
+                return ""
+            }
+
+            val content = file.readText()
+            val functionRegex = when (language) {
+                "kotlin" -> Regex("""fun\s+$functionName\s*\(.*\)\s*\{([\s\S]*?)\}""")
+                "java" -> Regex(
+                    """(?:public|private|protected)?\s+[\w<>\[\]]+\s+$functionName\s*\([^)]*\)\s*\{([\s\S]*?)}\s*""",
+                    RegexOption.MULTILINE,
+                )
+
+                else -> throw IllegalArgumentException("Unsupported language $language while extracting code section.")
+            }
+
+            val matchResult = functionRegex.find(content) ?: return ""
+            val codeSection = matchResult.groupValues[1].lines()
+            val filteredLines = mutableListOf<String>()
+            var capture = false
+
+            for (line in codeSection) {
+                if (docsStartPattern.containsMatchIn(line)) {
+                    capture = true
+                    continue
+                }
+                if (docsEndPattern.containsMatchIn(line)) {
+                    capture = false
+                    continue
+                }
+                if (capture) {
+                    filteredLines.add(line)
+                }
+            }
```

I think we can simplify this to avoid `continue`s:
```suggestion
            for (line in codeSection) {
                when {
                    docsStartPattern.containsMatchIn(line) -> capture = true
                    docsEndPattern.containsMatchIn(line) -> capture = false
                    capture -> filteredLines.add(line)
                }
            }
```